x-pack/plugins/ml/public/application/jobs/jobs_list/components/delete_job_modal/delete_job_modal.tsx

```diff
@@ -58,13 +58,18 @@ export const DeleteJobModal: FC<Props> = ({ setShowFunction, unsetShowFunction,
   }, []);
 
   const showModal = useCallback((jobs: MlSummaryJob[]) => {
-    setJobIds(jobs.map(({ id }) => id));
-    setHasManagedJob(jobs.some((job) => isManagedJob(job)));
+    setAdJobs(jobs);
     setModalVisible(true);
     setDeleting(false);
     setDeleteUserAnnotations(false);
   }, []);
 
+  const jobIds = adJobs.map(({ id }) => id);
+  const hasManagedJob = adJobs.some((job) => isManagedJob(job));
+  const hasAlertingRules = adJobs.some(
+    (job) => Array.isArray(job.alertingRules) && job.alertingRules.length > 0
+  );
```

Comment:
Nit: This whole block could go into a `useMemo` and return `const { jobIds, hasManagedJob, hasAlertingRules} = ...`. This will make sure that `jobIds` further down works correctly as the memoization check for the `closeModal`'s `useCallback`.

Reply:
updated in https://github.com/elastic/kibana/pull/176049/commits/ae2b3ae1bdbe848038ecd9ff456dbea0cb861d9e